dockers/docker-orchagent/supervisord.conf

```diff
@@ -59,35 +59,35 @@ stderr_logfile=syslog
 dependent_startup=true
 dependent_startup_wait_for=portsyncd:running
 
-[program:swssconfig]
-command=/usr/bin/swssconfig.sh
+[program:neighsyncd]
+command=/usr/bin/neighsyncd
 priority=5
 autostart=false
-autorestart=unexpected
-startretries=0
-startsecs=0
+autorestart=false
 stdout_logfile=syslog
 stderr_logfile=syslog
 dependent_startup=true
 dependent_startup_wait_for=orchagent:running
 
-[program:restore_neighbors]
-command=/usr/bin/restore_neighbors.py
+[program:swssconfig]
+command=/usr/bin/swssconfig.sh
 priority=6
 autostart=false
-autorestart=false
-startsecs=0
+autorestart=unexpected
 startretries=0
+startsecs=0
 stdout_logfile=syslog
 stderr_logfile=syslog
 dependent_startup=true
-dependent_startup_wait_for=swssconfig:exited
+dependent_startup_wait_for=neighsyncd:running
```

Comment:
Not sure that neighsyncd:running state does mean that neighsyncd already called AppRestartAssist::registerAppTable().
Potentially there is still a risk for this race condition to happen.
Probably AppRestartAssist::registerAppTable should be fixed - instead of clearing pending data let the orch consume it?
The AppRestartAssist is related to warm logic so such change may affect the warm reboot/restart.
@yxieca @prsunny @tahmed-dev @lguohan

Reply:
This issue is caused by the race condition between the init table and start writing table on two threads. I propose to put the init operation into swssconfig. It does not need to care what the detailed implementation of the underlying 'running' state.

Reply:
I agreed with @stepanblyschak. Could we add a check for warm-reboot at the beginning of AppRestartAssist if it's only needed for warm-reboot?